front/lib/api/assistant/feedback.ts

```diff
@@ -196,45 +147,106 @@ export async function deleteMessageFeedback(
     conversation: ConversationType | ConversationWithoutContentType;
     user: UserType;
   }
-): Promise<boolean | null> {
+) {
   const owner = auth.workspace();
   if (!owner) {
-    throw new Error("Unexpected `auth` without `workspace`.");
+    return new Err({
+      type: "workspace_not_found",
+      message: "The workspace you're trying to access was not found.",
+    });
   }
 
-  const message = await Message.findOne({
-    where: {
-      sId: messageId,
-      conversationId: conversation.id,
-    },
-    attributes: ["agentMessageId"],
-  });
-
-  if (!message || !message.agentMessageId) {
-    return null;
+  if (!canAccessConversation(auth, conversation)) {
+    return new Err({
+      type: "conversation_access_restricted",
+      message: "You don't have access to this conversation.",
+    });
   }
 
-  const agentMessage = await AgentMessage.findOne({
-    where: {
-      id: message.agentMessageId,
-    },
-  });
+  const feedbackWithContext =
+    await AgentMessageFeedbackResource.getFeedbackWithConversationContext({
+      auth,
+      messageId,
+      conversation,
+      user,
+    });
 
-  if (!agentMessage) {
-    return null;
+  if (feedbackWithContext.isErr()) {
+    return feedbackWithContext;
   }
 
-  const feedback =
-    await AgentMessageFeedbackResource.fetchByUserAndAgentMessage({
-      user,
-      agentMessage,
-    });
+  const { feedback } = feedbackWithContext.value;
 
   if (!feedback) {
-    return null;
+    return new Ok(undefined);
   }
 
-  const deletedFeedback = await feedback.delete(auth);
+  const deleteRes = await feedback.delete(auth, {});
+
+  if (deleteRes.isErr()) {
+    return deleteRes;
+  }
+
+  return new Ok(undefined);
+}
+
+export async function getAgentFeedbacks({
+  auth,
+  agentConfigurationId,
+  withMetadata,
+  filters,
+}: {
+  auth: Authenticator;
+  withMetadata: boolean;
+  agentConfigurationId: string;
+  filters?: {
+    limit?: number;
+    olderThan?: Date;
+    earlierThan?: Date;
+  };
+}): Promise<
+  Result<
+    (AgentMessageFeedbackType | AgentMessageFeedbackWithMetadataType)[],
+    Error
+  >
+> {
+  const owner = auth.workspace();
+  if (!owner || !auth.isUser()) {
+    return new Err(new Error("workspace_not_found"));
+  }
+  const plan = auth.plan();
+  if (!plan) {
+    return new Err(new Error("plan_not_found"));
+  }
+
+  // Make sure the user has access to the agent
+  const agentConfiguration = await getAgentConfiguration(
+    auth,
+    agentConfigurationId
+  );
+  if (!agentConfiguration) {
+    return new Err(new Error("agent_configuration_not_found"));
+  }
+
+  const feedbacksRes = await AgentMessageFeedbackResource.fetch({
```

Comment:
Let's avoid all uses of ModelId. `fetch` if defined by you sholud take the WorkspaceType